gpii/node_modules/couchConnector/src/couchConnector.js

```diff
@@ -0,0 +1,116 @@
+/*!
+
+    Convenience grade to provision and manage a CouchDB harness provided by the gpii-couchdb-test-harness package.
+    Intended as a replacement for the legacy "Pouch Manager".
+
+    Copyright 2018 Raising the Floor International.
+
+    Licensed under the New BSD license. You may not use this file except in compliance with this License.
+
+    You may obtain a copy of the License at
+    https://github.com/GPII/universal/blob/master/LICENSE.txt
+
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+/*
+
+    A "couch connector" that provides a persistent CouchDB harness that can be explicitly reset as needed.
+
+*/
+fluid.defaults("gpii.couchConnector", {
+    gradeNames: ["fluid.component"],
+    resetServerPort: 8060,
+    // Provided for (temporary) backward compatibility with the legacy pouch test harness.
+    databases: "{that}.options.pouchConfig.databases",
+    components: {
+        harness: {
+            type: "gpii.test.couchdb.harness.persistent",
+            options: {
+                databases: "{couchConnector}.options.databases",
+                listeners: {
+                    "onStartupComplete.escalate": "{couchConnector}.events.onCouchStartupComplete",
+                    "onDbProvisioningComplete.escalate": "{couchConnector}.events.onCouchProvisioningComplete"
+                }
+            }
+        },
+        expressReset: {
```

Comment:
I don't think that this component and its events need to have "express" in the name, since that's just a particularly library used to host an HTTP endpoint? Why do we address this over HTTP rather than just exposing a direct function call?

Reply:
This reset database functionality is not for the internal code use. It's for the convenience that someone who runs GPII application in the dev config, such as by running `NODE_ENV=gpii.config.untrusted.development.all.local.json5 node gpii.js` can easily reset the pouch/couch to the initial data set via this endpoint.

Without this reset, the updated docs to the backend pouchDB from separate GPII app runs will be maintained in the database.

See [the doc of PouchManager](https://github.com/GPII/universal/blob/master/documentation/PouchManager.md).

Reply:
@cindyli - still not clear why this couldn't just be a function call and still runnable from a config?

Reply:
@amb26 do you mean "using a function call" is, users can run a command such as `node reset-couch.js` to reset the database to the initial data set, instead of accessing /reset-couch endpoint? This `reset-couch.js` calls the function to perform reset?

Can you clarify the reset workflow by using a function call?

Reply:
Hi @cindyli - yes, that's what I mean - why is /reset-couch provided as an HTTP endpoint?

Reply:
Umm.. I don't remember why it was decided to use an endpoint, probably because I was deeply immersed in the pouchHarness express world at then. :)

I'm totally fine with using a node command to reset database.

Reply:
Bump this comment - I think we can axe the "express reset server"